### PR TITLE
Season switch2

### PR DESCRIPTION
**Pull request by F3PiX**
This replaces PR #309, and open todo's from  PR #304 
Adding tests. 
Optimize :back_to_reality option in SeasonsController

### CODE REVIEW

File: spec/models/season_spec.rb [hunk 1]
```diff
@@ -190,4 +190,30 @@
     end
   end
 
-end
+  ######## Testing Season Switch #############
+  describe 'switch phase' do
```
Comment (carpodaster) [inline spec/models/season_spec.rb]: Can you turn that outer `describe` into a `context` and then wrap the inner `it` blocks into a `describe #method_name` ?

Example:
```ruby
context 'switching phases' do
  describe '#fake_application_phase' do
    it 'timeshifts the application phase to today' do
      # ...
    end
  end

  describe '#fake_coding_phase' do
    # ...
  end
end
```

File: app/models/season.rb [hunk 2]
```diff
@@ -46,8 +46,20 @@ def ended?
 
   def year; name end
 
-  # switch_phase, in dev env only: enables developers to easily switch
+  # switch_phase: enables developers to easily switch
   # between time dependent settings in views
+  def fake_proposals_phase
+    update({
```
Comment (carpodaster) [inline app/models/season.rb]: You can use Rails' `from_now` so that e.g. `Date.today+6.months` becomes `6.months.from_now`

**Comment by carpodaster** [comment]
Awesome – thanks @F3PiX !

**Comment by alicetragedy** [comment]
:tada:

**Comment by F3PiX** [comment]
Thanks to you too :-)

**Pull request merged**